Products/CMFPlone/recyclebin.py

```diff
@@ -0,0 +1,584 @@
+from datetime import datetime
+from datetime import timedelta
+from persistent.mapping import PersistentMapping
+from plone.registry.interfaces import IRegistry
+from Products.CMFPlone.controlpanel.browser.recyclebin import (
+    IRecycleBinControlPanelSettings,
+)
+from Products.CMFPlone.interfaces.recyclebin import IRecycleBin
+from zope.annotation.interfaces import IAnnotations
+from zope.component import getUtility
+from zope.component.hooks import getSite
+from zope.interface import implementer
+
+import logging
+import uuid
+
+
+logger = logging.getLogger("Products.CMFPlone.RecycleBin")
+
+ANNOTATION_KEY = "Products.CMFPlone.RecycleBin"
+
+
+@implementer(IRecycleBin)
+class RecycleBin:
+    """Stores deleted content items"""
+
+    def __init__(self, context=None):
```

Comment:
This is trying to be some weird hybrid of a utility and an adapter. Pick one. Either make it an adapter that always takes the context as an arg when the object is constructed, or make it a utility with no args and always get the context using getSite

Reply:
changed iin https://github.com/plone/Products.CMFPlone/pull/4168/commits/712add76f8e6c028c147f3b282d2bedb76e22542